README.md

```diff
@@ -212,6 +212,40 @@ let coffeeMaker = Container.get<CoffeeMaker>("coffee.maker");
 coffeeMaker.make();
 ```
 
+### Using factory function to create service
+
+You can register your service with the container using factory function.
+
+This way, service instance will be created by calling your factory function instead of
+instantiating a class directly.
+
+```typescript
+import {Container} from "typedi";
+
+class Service {
+  constructor (public message: string) {
+  }
+}
+
+function createService () {
+  return new Service("Built with factory!");
+}
+
+// Registering service with container.
+// You can also pass additional arguments for your factory function.
+Container.registerService("my-service", Service, [], createService);

```

Comment:
Yeah, we will need to make it work first. Then I will be glad to update the docs.